File: tests/testdata/codegen/traverse-union-repro-pp/nodejs/traverse-union-repro.ts
import * as khulnasoft from "@khulnasoft/khulnasoft";
import * as aws from "@khulnasoft/aws";

const test = new aws.fsx.OpenZfsFileSystem("test", {
    storageCapacity: 64,
    subnetIds: [aws_subnet.test1.id],
    deploymentType: "SINGLE_AZ_1",
    throughputCapacity: 64,
});

File: tests/testdata/codegen/traverse-union-repro-pp/go/traverse-union-repro.go
package main

import (
	"github.com/khulnasoft/khulnasoft-aws/sdk/v5/go/aws/fsx"
	"github.com/khulnasoft/khulnasoft/sdk/v3/go/khulnasoft"
)

func main() {
	khulnasoft.Run(func(ctx *khulnasoft.Context) error {
		_, err := fsx.NewOpenZfsFileSystem(ctx, "test", &fsx.OpenZfsFileSystemArgs{
			StorageCapacity: khulnasoft.Int(64),
			SubnetIds: khulnasoft.String{
				aws_subnet.Test1.Id,
			},
			DeploymentType:     khulnasoft.String("SINGLE_AZ_1"),
			ThroughputCapacity: khulnasoft.Int(64),
		})
		if err != nil {
			return err
		}
		return nil
	})
}
